### PR TITLE
Move HRAM symbols to an assembled section

### PR DESCRIPTION
**Pull request by PikalaxALT**
The purpose of this update is to force HRAM symbols to output to the SYM and MAP files.  In the process, any erroneous references to HRAM symbols will be repaired.

### CODE REVIEW

File: engine/battle_anims/functions.asm [hunk 1]
```diff
@@ -1621,7 +1621,7 @@ Functioncd913: ; cd913 (33:5913)
 	ld hl, BATTLEANIMSTRUCT_10
 	add hl, bc
 	ld e, [hl]
-	ld hl, hTransferVirtualOAM ; $ff80
+	ld hl, -$80
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply engine/battle_anims/functions.asm]: 1) This is a signed int16.
2) It's not supposed to be an address in the first place.

**Comment by PikalaxALT** [comment]
The uncommitted file `constants/hram_constants.asm` has been deprecated.

File: engine/battle_anims/functions.asm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rangi42) [reply engine/battle_anims/functions.asm]: Oops, I wasn't paying attention. Thanks.

**Comment by Rangi42** [comment]
@PikalaxALT Can you please resolve the conflict with engine/overworld/map_objects.asm so this can hopefully get merged?

**Comment by Rangi42** [comment]
@PikalaxALT Can you please resolve the conflicts with this PR so it can hopefully get merged?

**Comment by mid-kid** [comment]
Just going to leave my generic upboat here for exposure.
It only requires a few changes to get approval from me:
- Rebase to current pokecrystal
- Make sure `ldh` is used consistently for hardware regs as well (i.e. enable the `-L` option for rgbasm in the makefile to make compare and make sure it's consistent, remove afterwards).

**Comment by Rangi42** [comment]
@PikalaxALT I can merge this if you fix the conflicting files.

**Comment by Rangi42** [comment]
PikalaxALT doesn't plan to maintain this PR.

**Pull request closed without merging**